libgreat/host/pygreat/board.py

```diff
@@ -352,6 +353,14 @@ def read_debug_ring(self, max_length=4096, clear=False, encoding='latin1'):
 
 def _to_hex_string(byte_array):
     """Convert a byte array to a hex string."""
+
+    # Python2 compatibility
+    try:
+        byte_array = future.builtins.bytes(byte_array)
+    except ValueError:
+        byte_array = list(byte_array)
```

Comment:
I can't run this code to check the possible types of `byte_array`, but I suspect from reading the diff that the `byte_array` argument might be a `str` on 2.7 and `bytes` on 3, or maybe a `bytearray` on either.  If that's true, it can likely be written without conditional code using [`bytearray`](https://docs.python.org/2.7/library/functions.html#bytearray):

```
byte_array = bytearray(byte_array)
```

I'd expect something called a `byte_array` to be the `bytearray` type.  Consider renaming the argument if it's not.